packages/babel-plugin-proposal-object-rest-spread/test/fixtures/object-spread/side-effect/output.js

```diff
@@ -0,0 +1,39 @@
+function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }
+
+function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }
+
+function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
+
+var k = {
+  a: 1,
+  b: 2
+};
+
+var o = _objectSpread(_objectSpread({
+  a: 3
+}, k), {}, {
+  b: k.a++
+});
+
+var pureA = {};
+var pureB = {};
+var pureC = {};
+var pureD = {};
+var pureE = {};
+
+function impureFunc() {
+  console.log('hello');
+}
+
+var output = _objectSpread(_objectSpread(_objectSpread({}, pureA), {}, {
+  get foo() {},
+
+  get bar() {}
+
+}, pureB, {}, pureC, {}), impureFunc(), {}, pureD, {
+  pureD
+});
```

Comment:
I would expect this output to be
```js
var output = { ...pureA, get foo() {}, get bar() {}, ...pureB, ...pureC, ...impureFunc(), ...pureD, pureD }

var output = _objectSpread(
  _objectSpread(
    {},
    pureA,
    { get foo() {}, get bar() {} },
    pureB,
    {},
    pureC,
    {}
  ),
  impureFunc(),
  {},
  pureD,
  { pureD }
);
```

I think that it's a bug in the `isPure` implementation, that only handles `isClassMethod` instead of `isMethod` (that also catches object methods/accessors).

Reply:
🙆‍♂️ I'm gonna give it a try.